core/src/apps/ethereum/layout.py

```diff
@@ -110,12 +111,11 @@ def require_confirm_data(ctx: Context, data: bytes, data_total: int) -> Awaitabl
 
 
 async def confirm_typed_data_final(ctx: Context) -> None:
-    await confirm_text(
+    await confirm_action(
         ctx,
         "confirm_typed_data_final",
         title="Confirm typed data",
-        data="",
-        description="Confirm signing EIP-712 typed data?",
+        action="Confirm signing\nEIP-712 typed data?",
```

Comment:
didn't confirm_action auto-wrap the text?

Reply:
In my opinion, manually wrapping it looks better, but I'm happy to let it auto-wrap if that's preferable.

(for some reason the "p" button didn't take a screenshot, so here's a manual screenshot)
![image](https://user-images.githubusercontent.com/19716675/149359065-cc73f63d-4a6f-4785-95c8-ff8087a0e10f.png)